include/crocoddyl/core/numdiff/diff-action.hxx

```diff
@@ -99,22 +141,55 @@ void DifferentialActionModelNumDiffTpl<Scalar>::calcDiff(const boost::shared_ptr
     model_->calc(d->data_u[iu], x, u + d->du);
 
     const VectorXs& xn = d->data_u[iu]->xout;
-    const Scalar c = d->data_u[iu]->cost;
+    const Scalar cp = d->data_u[iu]->cost;
     data->Fu.col(iu) = (xn - xn0) / disturbance_;
 
-    data->Lu(iu) = (c - c0) / disturbance_;
+    data->Lu(iu) = (cp - c0) / disturbance_;
     d->Ru.col(iu) = (d->data_u[iu]->r - d->data_0->r) / disturbance_;
+
+    // We can apply the same formulas for finite difference as above
+
+    // d->du(iu) = -disturbance_; 
+    model_->calc(d->data_u[iu], d->xp, u- d->du);
+    data->Luu(iu,iu) = (d->data_u[iu]->cost + cp - 2*c0)/ (disturbance_*disturbance_);    
+    std::cout<<"Here is the value of iu :"<<iu<<"\n";
+
+    for (std::size_t ju = iu+1; ju < model_->get_nu(); ++ju) {
+      
+      std::cout<<"Here is the value of ju :"<<ju<<"\n";
+
+      d->du(ju) = disturbance_;
+
+      model_->calc(d->data_u[iu], d->xp, u + d->du);
+      const Scalar c_pp = d->data_u[iu]->cost;  //cost due to positive disturbance in both directions
+      std::cout<<"Here is the value of du :"<<d->du<<"\n";
+
+      d->du(iu) = 0.0;
+
+      model_->calc(d->data_u[iu], d->xp, u + d->du);
+      const Scalar c_zp = d->data_u[iu]->cost;  //cost due to zero disturance in 'i' and positive disturbance in 'j' direction
+      std::cout<<"Here is the value of du :"<<d->du<<"\n";
+      std::cout<<"Here is the value of c,  c_pp, c_zp :"<<cp << c_pp << c_zp<<"\n";
+    
+      data->Luu(iu,ju) = (c_pp - c_zp - cp + c0)/ (disturbance_*disturbance_);
+      
+      data->Luu(ju,iu) = data->Luu(iu,ju); // can also be implemented at the end by Luu = Luu.T
+      d->du(iu) = disturbance_;
+      d->du(ju) = 0.0;
+    }
+
     d->du(iu) = 0.0;
+
   }
 
   if (with_gauss_approx_) {
-    data->Lxx = d->Rx.transpose() * d->Rx;
+    // data->Lxx = d->Rx.transpose() * d->Rx;
     data->Lxu = d->Rx.transpose() * d->Ru;
-    data->Luu = d->Ru.transpose() * d->Ru;
+    // data->Luu = d->Ru.transpose() * d->Ru;
   } else {
-    data->Lxx.setZero();
+    // data->Lxx.setZero();
     data->Lxu.setZero();
-    data->Luu.setZero();
+    // data->Luu.setZero();
   }
```

Comment:
It seems you forget to remove these lines. 

As in the action case, I would remove the Rx, Ru computations and also deprecate the "with_gauss_approx" argument in the constructor.